docker/generic/start_proxy.py

```diff
@@ -731,7 +741,11 @@ def gen_proxy_config(args):
         proxy_conf.extend(["--v", "0"])
 
     if args.envoy_xff_num_trusted_hops:
-         proxy_conf.extend(["--envoy_xff_num_trusted_hops", args.envoy_xff_num_trusted_hops])
+        proxy_conf.extend(["--envoy_xff_num_trusted_hops",
+                           args.envoy_xff_num_trusted_hops])
+    elif args.on_serverless:
+        proxy_conf.extend(["--envoy_xff_num_trusted_hops",
```

Comment:
Logic wise, I feel this logic should be part of serverless/env_start_proxy.py.   We don't want to move that logic here.
Imaging,  we may have another deploying mode: xyz  calling start_proxy.py,   we don't want to move its logic into here too.

Reply:
I considered this, but it would result in a lot of code duplication. 

We need to account for the case where the user wants to override `envoy_xff_num_trusted_hops`. If we set it to 0 directly in `env_start_proxy.py`, then we need to add another flag parser to allow the user to override it.

So instead we just signal what platform we're on and pass all the user's flags directly from `env_start_proxy` to `start_proxy`. Since `start_proxy` has the `args_parser`, it's much easier to handle here.

Reply:
Ok,  I see your point